spec/requests/cart/add_item_to_a_cart_spec.rb

```diff
@@ -0,0 +1,42 @@
+require "rails_helper"
+
+RSpec.describe "Cart", type: :request do
+  describe "Add item to user cart" do
+    let(:user) { create(:user) }
+    let(:book) { create(:book) }
+
+    def add_item_to_cart(book_id)
+      post cart_path, params: {
+        cart: {
+          book_id: book_id
+        }
+      }
+    end
+
+    def login(email, password)
+      post login_path, params: {
+        user: {
+          email: email,
+          password: password,
+        }
+      }
+    end
+
+    context "when the user does not login" do
+      it "will fail" do
+        add_item_to_cart(10)
+        expect(response).to have_http_status(:found)
+      end
+    end
+
+    context "when the use login" do
+      it "will succeeded" do
+        login(user.email, user.password)
+        expect(response).to have_http_status(:redirect)
+        add_item_to_cart(book.id)
+        expect(response).to have_http_status(:ok)
+        expect(user.cart.orders.count).to eq 1
```

Comment:
You can also write like this:
```suggestion
        expect {
          add_item_to_cart(book.id)
          expect(response).to have_http_status(:ok)
        }.to change { user.cart.orders.count }.by(1)
```

Reply:
Could we also verify that the book ID in the cart is correct?

Reply:
Yes, we can. Could you review my idea?
```suggestion
        expect {
          add_item_to_cart(book.id)
          expect(response).to have_http_status(:ok)
        }.to change { user.cart.orders.count }.by(1)
        book_is_added_to_cart = user.cart.books.last
        expect(book_is_added_to_cart).to eq book
```

Reply:
The above code looks good to me